### ISSUE TITLE
Issues with logics in openjdk*-pipeline publish process

### ISSUE THREAD

**Issue by sophia-guo** [adoptium/ci-jenkins-pipelines#1148]
Currently the openjdk*-pipeline can be triggered manually with releaseType='weekly' and enableTests=false. That is with no tests at all all build results will be published to temurin binary repo. When run tests query by nightly or url directly tests failed without workable jdk. https://github.com/adoptium/temurin8-binaries/releases/download/jdk8u442-b02-ea-beta/OpenJDK8U-jdk_x64_mac_hotspot_8u442b02-ea.tar.gz.

Another thing is in openjdk*-pipeline build it's clearly build job failed, we will still publish archived artifacts. https://ci.adoptium.net/job/build-scripts/job/openjdk8-pipeline/2947/consoleFull. Four platforms build jobs failed, artifacts published regardless. This is common to release build, as it's difficult to tell if the build job itself (build process ) fails or it fails because the downstream test jobs fail.

**Comment by andrew-m-leonard** [comment]
> 
> Another thing is in openjdk*-pipeline build it's clearly build job failed, we will still publish archived artifacts. https://ci.adoptium.net/job/build-scripts/job/openjdk8-pipeline/2947/consoleFull. Four platforms build jobs failed, artifacts published regardless. This is common to release build, as it's difficult to tell if the build job itself (build process ) fails or it fails because the downstream test jobs fail. Need to think about discriminating those two situation and decide publish or not.

I'm thinking we should try and get to the position of not publishing if the build job status is not SUCCESSFUL|UNSTABLE,
we still unfortunately have some test jobs that fail...we should bite the bullet and make that change maybe?? force us to fix those failures..........

**Comment by sophia-guo** [comment]
Another choice is not publishing if build itself failed ( build, any other build related jobs - sign, installer,..). This can be done by check job status ( set by related jobs) before tests. If it's non successful  do the artifacts deletion so no artifacts available. Hence no artifacts will be publish for EA and no release link will be generated for release.

**Comment by andrew-m-leonard** [comment]
The discussion point needs to be had about whether we still publish EA builds regardless of test results.

**Comment by sophia-guo** [comment]
Yes, there are two cases needed to discuss. I'm thinking for the build failed itself we shouldn't publish by deleting the artifacts.

- build job failed itself ( build, any other build related jobs - sign, installer,... etc)
- build job failed due to test failures.
